docker/docker-entrypoint.sh

```diff
@@ -3,37 +3,41 @@
 set -e
 
 # Get MuJoCo key from the environment
-echo "${MJKEY}" > /root/.mujoco/mjkey.txt
+echo -e "${MJKEY}" > /root/.mujoco/mjkey.txt
 
-# Setup dummy X server display
-display_num=0
-export DISPLAY=:"${display_num}"
-Xvfb "${DISPLAY}" -screen 0 1024x768x24 &
-pulseaudio -D --exit-idle-time=-1
+if [[ "${IMAGE_TYPE}" == "headless" ]]; then
+  # Setup dummy X server display
+  # Socket for display :0 may already be in use if the container is connected
+  # to the network of the host, that's why we use 1.
+  display_num=1
+  export DISPLAY=:"${display_num}"
+  Xvfb "${DISPLAY}" -screen 0 1024x768x24 &
+  pulseaudio -D --exit-idle-time=-1
 
-# Wait for X to come up
-file="/tmp/.X11-unix/X${display_num}"
-for i in $(seq 1 10); do
+  # Wait for X to come up
+  file="/tmp/.X11-unix/X${display_num}"
+  for i in $(seq 1 10); do
     if [ -e "$file" ]; then
       break
     fi
     echo "Waiting for X to start (i.e. $file to be created) (attempt $i/10)"
     sleep "$i"
-done
-if ! [ -e "$file" ]; then
+  done
+  if ! [ -e "$file" ]; then
     echo "Timed out waiting for X to start: $file was not created"
     exit 1
+  fi
+  export DISABLE_MUJOCO_RENDERING=1  # silence glfw
 fi
 
 # Activate conda environment
 source activate garage
 
 # Fixes Segmentation Fault
 # See: https://github.com/openai/mujoco-py/pull/145#issuecomment-356938564
-export LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libGLEW.so:/usr/lib/x86_64-linux-gnu/mesa/libGL.so.1
+export LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libGLEW.so
```

Comment:
does this work on nvidia? you would need to preload the nvidia version.

Reply:
It does work on nvidia.